react-native-purchases-ui/android/src/main/java/com/revenuecat/purchases/react/ui/FontAssetManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.revenuecat.purchases.react.ui
+
+import android.content.res.AssetManager
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import com.facebook.infer.annotation.Nullsafe
+
+@Nullsafe(Nullsafe.Mode.LOCAL)
+object FontAssetManager {
+    private val mFontCache: MutableMap<String?, List<Font>?> = hashMapOf()
+    private val EXTENSIONS = arrayOf("", "_bold", "_italic", "_bold_italic")
+    private val FILE_EXTENSIONS = arrayOf(".ttf", ".otf")
+
+    private fun getStyleFromExtension(style: Int): FontStyle {
+        return when (style) {
+            2, 3 -> FontStyle.Italic
+            else -> FontStyle.Normal
+        }
+    }
+
+    private fun getWeightFromExtension(style: Int): FontWeight {
+        return when (style) {
+            1, 3 -> FontWeight.Bold
+            else -> FontWeight.Normal
+        }
+    }
+
+    fun getFontList(
+        fontFamilyName: String,
+        assetManager: AssetManager
+    ): List<Font> {
+        var assetFontFamily = mFontCache[fontFamilyName]
+        if (assetFontFamily == null) {
+            assetFontFamily = mutableListOf()
+            mFontCache[fontFamilyName] = assetFontFamily
+            for ((style, styleExtension) in EXTENSIONS.withIndex()) {
+                for (fileNameExtension in FILE_EXTENSIONS) {
+                    val fileName = "fonts/$fontFamilyName$styleExtension$fileNameExtension"
```

Comment:
Should we check if the file exists before creating the `Font`? Checking the API, seems that it won't be nullable, so not sure if that may affect the behavior or not

Reply:
In the ReactFontManager, it looks like it's wrapped around a try catch. maybe we could do the samething for now

Reply:
ok so I made the change here https://github.com/RevenueCat/react-native-purchases/pull/902/commits/4c23c9a6144587e54c9377348cbf58f889ca6b70

but I'm torn between falling back to default or throwing an error
```kotlin
 
    if (fontList.isEmpty()) {
        throw NoSuchElementException("No fonts found for font family: $fontFamilyName")
    }
```
or returning a null like it is now
```kotlin
return if (fontList.isNotEmpty()) {
            val fontFamily = FontFamily(fontList)
            fontFamilyHashMap[fontFamilyName] = fontFamily
            fontFamily
        } else {
            null
        }
```

Reply:
I believe returning null is the better option... Ideally in case it can't be found, we just log an error and continue, using the default font.